### ISSUE TITLE
ValueError: Illegal argument: host=::1

### ISSUE THREAD

**Issue by gryphius** [jorritfolmer/TA-dmarc#26]
I was trying out this addon on our company splunk instance, but just the error below after setting up an IMAP account to pull dmarc reports from

```python
2020-06-04 07:37:11,822 ERROR pid=3137 tid=MainThread file=base_modinput.py:log_error:307 | Traceback (most recent call last):
  File "/opt/splunk/etc/apps/TA-dmarc/bin/ta_dmarc/modinput_wrapper/base_modinput.py", line 113, in stream_events
    self.parse_input_args(input_definition)
  File "/opt/splunk/etc/apps/TA-dmarc/bin/ta_dmarc/modinput_wrapper/base_modinput.py", line 152, in parse_input_args
    self._parse_input_args_from_global_config(inputs)
  File "/opt/splunk/etc/apps/TA-dmarc/bin/ta_dmarc/modinput_wrapper/base_modinput.py", line 170, in _parse_input_args_from_global_config
    global_config = GlobalConfig(uri, session_key, global_schema)
  File "/opt/splunk/etc/apps/TA-dmarc/bin/ta_dmarc/splunktaucclib/global_config/__init__.py", line 51, in __init__
    port=splunkd_info.port,
  File "/opt/splunk/etc/apps/TA-dmarc/bin/ta_dmarc/solnlib/net_utils.py", line 129, in wrapper
    'Illegal argument: {}={}'.format(arg, value))
ValueError: Illegal argument: host=::1
```

From the traceback it looks to me like the addon (or one of its dependencies) does not support IPv6 addresses?

**Comment by jorritfolmer** [comment]
Thanks for your report! I've been trying to reproduce your finding, but no luck so far.
For testing I:
1. Used RHEL 7.4, Splunk 8.0.0, TA-dmarc 3.2.1
1. setup an IMAP server on localhost with Dovecot 
(RHEL 7x, Splunk 8.0.0, 
1. configured the add-on to connect to `::1`

It works, these are the logs:

```
2020-06-05 08:59:17,589 INFO pid=41504 tid=MainThread file=base_modinput.py:log_info:293 | Start processing imap server ::1 with use_ssl True
2020-06-05 08:59:17,712 INFO pid=41504 tid=MainThread file=base_modinput.py:log_info:293 | get_dmarc_messages: 12 messages in folder INBOX
2020-06-05 08:59:17,753 INFO pid=41504 tid=MainThread file=base_modinput.py:log_info:293 | get_dmarc_messages: 0 messages in folder INBOX match subject "Report domain:"
2020-06-05 08:59:17,754 INFO pid=41504 tid=MainThread file=base_modinput.py:log_info:293 | Ended processing imap server ::1
```

Your logs indicate some exception in the splunk libs that the addon uses. Not sure what to make of it, or how to create circumstances to understand and fix it.

Can you share some details about your setup?

**Comment by gryphius** [comment]
We're running: 

Splunk Version 7.3.5
TA-dmarc	3.2.1
RHEL 7.8

Note, that i did *not* configure the addon to connect to an IMAP server on `::1`, I should have mentioned that more clearly. 
The value `::1` seems to be somewhere in the global splunk config, I do not know where exactly as I don't operate the splunk instance myself, but I'll try to find out.

**Comment by jorritfolmer** [comment]
This looks similar: https://answers.splunk.com/answers/729032/error-while-using-add-on-developed-using-add-on-bu.html

Can you check if one of those Python hacks in that thread fixes it for you?

**Comment by gryphius** [comment]
> Can you check if one of those Python hacks in that thread fixes it for you?

changing the the `is_valid_hostname` function in `bin/ta_dmarc/solnlib/net_utils.py` fixes the issue:

```python
allowed = re.compile('(?!-)[A-Z\d\-\:]{1,63}(?<!-)$', re.IGNORECASE)
````

Thanks for the pointer!

**Comment by jorritfolmer** [comment]
Great! This fix is included in release 3.2.2. I'll update on Splunkbase shortly.
